src/index.js

```diff
@@ -0,0 +1,139 @@
+import {
+  connection,
+  decoding,
+  status,
+  constants
+} from 'pipboylib';
+
+import {
+  Observable
+} from 'rx'
+
+const {
+  discover,
+  createSocket,
+  sendPeriodicHeartbeat,
+  createConnectionSubject
+} = connection
+
+const {
+  parseBinaryMap,
+  parseBinaryDatabase,
+  aggregateBundles,
+  generateTreeFromDatabase
+} = decoding
+
+const {
+  connected
+} = status
+
+const {
+  channels
+} = constants
+
+const pipboyColor = [
+  0.1,
+  1.0,
+  0.1
+];
+
+discover()
+  .then(server => createSocket(server.info.address))
+  .then(socket => {
+    sendPeriodicHeartbeat(socket)
+    return createConnectionSubject(socket)
+  })
+  .then(subject => {
+    connected(subject)
+      .then(handshake => {
+        // Create Canvas for Fallout 4 Map
+        const canvas = document.getElementById('map')
+        const context = canvas.getContext('2d')
+
+        const arrow = document.getElementById('arrow')
+        arrow.style.left = '50%'
+        arrow.style.top = '50%'
+
+        const localmap = subject
+          .filter(x => x.type === channels.LocalMapUpdate)
+          .map(x => parseBinaryMap(x.payload))
+          .distinctUntilChanged()
+
+        const resize = Observable
+          .fromEvent(window, 'resize')
+          .merge(Observable.just())
```

Comment:
Wow and full screen is awesome. We'll want to make sure we keep the map at the right scale later.